src/main/scala/intellij/haskell/action/CreateHaskellFileAction.scala

```diff
@@ -50,6 +56,54 @@ class CreateHaskellFileAction extends CreateFileFromTemplateAction(CreateHaskell
     })
   }
 
+
+  override def createFileFromTemplate(originName: String, template: FileTemplate, originDir: PsiDirectory): PsiFile = {
+    val pathItems = HaskellFileUtil.getPathFromSourceRoot(originDir.getProject, originDir.getVirtualFile).orNull
+
+    // Adapted from super definition.
+    val mkdirs = new CreateFileAction.MkDirs(originName, originDir)
+    val name = mkdirs.newName
+    val dir = mkdirs.directory
+    val project = dir.getProject
+
+    val nameWithmodulePrefix = if (pathItems == null || invalidPathItems(pathItems) || pathItems.isEmpty) {
+      name
+    } else {
+      pathItems.mkString(".") + "." + name
+    }
+
+    try {
+      // Patch props with custom property.
+      val props = FileTemplateManager.getInstance(project).getDefaultProperties()
+      props.setProperty("NAME", nameWithmodulePrefix)
+      val element = FileTemplateUtil.createFromTemplate(template, name, props, dir)
+
+      val psiFile = element.getContainingFile
+
+      val virtualFile = psiFile.getVirtualFile
+      if (virtualFile != null) {
+        FileEditorManager.getInstance(project).openFile(virtualFile, true)
+        val defaultTemplateProperty = getDefaultTemplateProperty
+        if (defaultTemplateProperty != null) {
+          PropertiesComponent.getInstance(project).setValue(defaultTemplateProperty, template.getName)
+        }
+        return psiFile
+      }
+    } catch {
+      case e: ParseException => Messages.showErrorDialog(project, "Error parsing Velocity template: " + e.getMessage, "Create File from Template");
```

Comment:
O, wait there is already one created by me but in Settings/Templates IntelliJ shows something else...